lib/charms/ovn_charm.py

```diff
@@ -729,6 +733,41 @@ def render_nrpe(self):
             charm_nrpe, self.nrpe_check_services, current_unit)
         charm_nrpe.write()
 
+    @staticmethod
+    def get_machine_id():
+        """Get local machine ID.
+
+        :returns: machine ID
+        :rtype: bytearray
+        :raises: OSError
+        """
+        with open('/etc/machine-id', 'r') as fin:
+            return bytearray.fromhex(fin.read().rstrip())
+
+    @staticmethod
+    def unique_bridge_mac(machine_id, bridge_name):
+        """Generate uniqe mac address for use on a bridge interface.
+
+        The bridge interface will be visible in the datapath and as such the
+        address we choose must be globally unique. We accomplish this by
+        composing a MAC address from the local machine-id(5), a prefix and the
+        name of the bridge.
+
+        :param machine_id: Local machine ID.
+        :type machine_id: bytearray
+        :param bridge_name: Name of bridge for which the address will be used.
+        :type bridge_name: str
+        :returns: String representation of generated MAC address.
+        :rtype: str
+        """
+        # initialize with 'fa:16:3e' prefix
+        generated = bytearray.fromhex('fa163e')
```

Comment:
This matches what is typically used for virtual machines etc. We should probably use a different prefix